Spring Boot intro 1.0

aksoli666 opened the pull request:
added checkstyle and the following property;
created a .github directory with workflows and ci.yml; created Book entity;
created BookRepository and BookRepositoryImpl;
created BookService and BookServiceImpl;
added CommandLineRunner bean to main class;

aksoli666 committed:
added checkstyle and the following property;
created a .github directory with workflows and ci.yml;
created Book entity;
created BookRepository and BookRepositoryImpl;
created BookService and BookServiceImpl;
added CommandLineRunner bean to main class;

Elena-Bruyako commented on src/main/java/shop/config/AppConfig.java:
```diff
@@ -0,0 +1,47 @@
+@Configuration
+@PropertySource("classpath:application.properties")
+@ComponentScan(basePackages = "shop")
+public class AppConfig {
```
remove class

Elena-Bruyako commented on src/main/java/shop/entity/Book.java:
```diff
@@ -0,0 +1,32 @@
+
+@Entity
+@Table(name = "books")
+@Data
```
```suggestion
@Getter
@Setter
```

Elena-Bruyako commented on src/main/java/shop/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+import shop.entity.Book;
+import shop.repository.BookRepository;
+
+@Repository
```
```suggestion
@Repository
@RequiredArgsConstructor
```

Elena-Bruyako commented on src/main/java/shop/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
```suggestion
```

Elena-Bruyako commented on src/main/java/shop/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new EntityExistsException("Book already exists or something is wrong. "
```
```suggestion
            throw new DataProcessingException("Can't save book: " + book, e");
```

Elena-Bruyako commented on src/main/java/shop/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("SELECT b FROM Book b", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Error finding all books", e);
```
```suggestion
            throw new DataProcessingException("Fetching books failed", e);
```

Elena-Bruyako commented on src/main/java/shop/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("SELECT b FROM Book b", Book.class).getResultList();
```
```suggestion
            return session.createQuery("FROM Book", Book.class).getResultList();
```

Elena-Bruyako commented on src/main/java/shop/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,28 @@
+import shop.repository.BookRepository;
+import shop.service.BookService;
+
+@Service
```
```suggestion
@Service
@RequiredArgsConstructor
```

Elena-Bruyako commented on src/main/java/shop/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,28 @@
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
```suggestion
```

Elena-Bruyako reviewed (changes requested):
see comments

aksoli666 committed:
Spring Boot intro 2.0
used @RequiredArgsConstructor;
created DataProcessingException;
changed @Data to @Setter and @Setter;
changed query in getAll() in BookRepositoryImpl;
